hpy/devel/include/hpy/hpytype.h

```diff
@@ -76,43 +88,68 @@ typedef struct {
 
        HPyType_HELPERS(PointObject)
 
+   It is also possible to inherit from some built-in types. The list of
+   available built-in base types is given in enum `HPyTupe_BuiltinShape`.
+   In case you want to inherit from one of those, it is necessary to specify
+   the base built-in type in the `HPyType_HELPERS` macro. Here is an example
+   for a pure HPy custom type inheriting from a built-in type 'tuple':
+
+       HPyType_HELPERS(PointObject, HPyType_BuiltinShape_Tuple)
+
    This would generate the following:
 
    * `PointObject * PointObject_AsStruct(HPyContext *ctx, HPy h)`: a static
      inline function that uses HPy_AsStruct to return the PointObject struct
      associated with a given handle. The behaviour is undefined if `h`
      is associated with an object that is not an instance of PointObject.
+     However, debug mode will catch an incorrect usage.
 
-   * `PointObject_IS_LEGACY`: an enum value set to 0 so that in the
-     HPyType_Spec for PointObject one can write
-     `.legacy = PointObject_IS_LEGACY` and not have to remember to
-     update the spec when the helpers used changes.
+   * `SHAPE(PointObject)`: a macro that is meant to be used as static
+     initializer in the corresponding HPyType_Spec. It is recommended to write
+     `.builtin_shape = SHPAE(PointObject)` such that you don't have to remember
```

Comment:
```suggestion
     `.builtin_shape = SHAPE(PointObject)` such that you don't have to remember
```

Reply:
btw. the reason why I now introduced `SHAPE(...)` is because (1) `g++` doesn't like if we assign an anonymous enum to `HPyType_BuiltinShape` and (2) because `MSVC` doesn't like `static const HPyType_BuiltinShape var` in static initializers.